subsys/bluetooth/mesh/net.c

```diff
@@ -442,18 +442,35 @@ static int net_header_encode(struct bt_mesh_net_tx *tx, uint8_t nid,
 	return 0;
 }
 
-static int net_encrypt(struct net_buf_simple *buf,
-		       const struct bt_mesh_net_cred *cred, uint32_t iv_index,
-		       bool proxy)
+int bt_mesh_encrypt_and_obfuscate(struct net_buf_simple *buf,
+				  const struct bt_mesh_net_cred *cred,
+				  uint32_t iv_index, bool proxy)
 {
+	struct bt_mesh_net_crypto_ctx ctx = {
```

Comment:
I'm now quite concern how much we gain from this change considering all this manipulations with stack for the sake of copying 24 bytes at most? I also see code size increased a bit (though this maybe not relevant characteristic for this change).

Reply:
I think the worst PR is also a conventional optimization. For messages that need relay, we will avoid copying twice. I think it is always good to avoid copying as much as possible. As for the increase of code size, further investigation is needed.

Reply:
I agree that is better to avoid copying. But in this particular case I'm struggling to see the benefit.

Reply:
yeap, i did build test, actually increase almost 300 bytes code size, ram size not change.

Reply:
The main question is open though...

Reply:
@jhedberg what do you think?

Reply:
@LingaoM is it possible to provide any metric that can be used as a proof of the improvement?